Libraries/LibWeb/ViewTransition/ViewTransition.cpp

```diff
@@ -0,0 +1,1046 @@
+                    backdrop-filter: {};
+                    color-scheme: {};
+                }}
+            )",
+                                                              transition_name, width, height, "transform", "writing_mode", "direction", "text_orientation", "mix_blend_mode", "backdrop_filter", "color_scheme")),
+                stylesheet->rules().length()));
+            // FIXME: all the strings above should be the identically named variables, serialized somehow.
+            captured_element->group_styles_rule = as<CSS::CSSStyleRule>(stylesheet->css_rules()->item(index));
+        }
+        // Otherwise, update capturedElement’s group styles rule to match the following CSS:
+        // :root::view-transition-group(transitionName) {
+        //   width: width;
+        //   height: height;
+        //   transform: transform;
+        //   writing-mode: writingMode;
+        //   direction: direction;
+        //   text-orientation: textOrientation;
+        //   mix-blend-mode: mixBlendMode;
+        //   backdrop-filter: backdropFilter;
+        //   color-scheme: colorScheme;
+        // }
+        // NOTE: The above code example contains variables to be replaced.
+        else {
+            captured_element->group_styles_rule->set_selector_text(MUST(String::formatted(":root::view-transition-group({0})", transition_name)));
+            captured_element->group_styles_rule->set_css_text(MUST(String::formatted(R"(
+                width: {};
+                height: {};
+                transform: {};
+                writing-mode: {};
+                direction: {};
+                text-orientation: {};
+                mix-blend-mode: {};
+                backdrop-filter: {};
+                color-scheme: {};
+            )",
+                width, height, "transform", "writing_mode", "direction", "text_orientation", "mix_blend_mode", "backdrop_filter", "color_scheme")));
+            // FIXME: all the strings above should be the identically named variables, serialized somehow.
+        }
+
+        // 5. If capturedElement’s new element is not null, then:
+        if (captured_element->new_element) {
+            // 1. Let new be the ::view-transition-new() with the view transition name transitionName.
+            ReplacedNamedViewTransitionPseudoElement* new_;
+            m_transition_root_pseudo_element->for_each_in_inclusive_subtree_of_type<ReplacedNamedViewTransitionPseudoElement>([&](auto& element) {
+                if (element.m_type == CSS::PseudoElement::ViewTransitionNew && element.m_view_transition_name == transition_name) {
+                    new_ = &element;
+                    return TraversalDecision::Break;
+                }
+                return TraversalDecision::Continue;
+            });
+
+            // 2. Set new’s replaced element content to the result of capturing the image of capturedElement’s
+            //    new element.
+            new_->m_content = captured_element->old_image;
```

Comment:
This doesn't seem to match the spec text?

Reply:
Indeed it does not.
Not sure how I missed that.